rasa/data.py

```diff
@@ -10,69 +10,73 @@
-    """Recursively collects all Core training files from a list of directories.
+def get_core_directory(paths: Union[Text, List[Text]]) -> Text:
+    """Recursively collects all Core training files from a list of paths.
 
     Args:
-        directories: List of paths to directories containing training files.
+        paths: List of paths to training files or folders containing them.
 
     Returns:
         Path to temporary directory containing all found Core training files.
     """
-    core_files, _ = _get_core_nlu_files(directories)
+    core_files, _ = _get_core_nlu_files(paths)
     return _copy_files_to_new_dir(core_files)
 
 
-def get_nlu_directory(directories: Union[Text, List[Text]]) -> Text:
-    """Recursively collects all NLU training files from a list of directories.
+def get_nlu_directory(paths: Union[Text, List[Text]]) -> Text:
+    """Recursively collects all NLU training files from a list of paths.
 
     Args:
-        directories: List of paths to directories containing training files.
+        paths: List of paths to training files or folders containing them.
 
     Returns:
         Path to temporary directory containing all found NLU training files.
     """
-    _, nlu_files = _get_core_nlu_files(directories)
+    _, nlu_files = _get_core_nlu_files(paths)
     return _copy_files_to_new_dir(nlu_files)
 
 
-def get_core_nlu_directories(directories: Union[Text, List[Text]]) -> Tuple[Text, Text]:
-    """Recursively collects all training files from a list of directories.
+def get_core_nlu_directories(paths: Union[Text, List[Text]]) -> Tuple[Text, Text]:
+    """Recursively collects all training files from a list of paths.
 
     Args:
-        directories: List of paths to directories containing training files.
+        paths: List of paths to training files or folders containing them.
 
     Returns:
         Path to directory containing the Core files and path to directory
         containing the NLU training files.
     """
-    story_files, nlu_data_files = _get_core_nlu_files(directories)
+    story_files, nlu_data_files = _get_core_nlu_files(paths)
 
     story_directory = _copy_files_to_new_dir(story_files)
     nlu_directory = _copy_files_to_new_dir(nlu_data_files)
 
     return story_directory, nlu_directory
 
 
-def _get_core_nlu_files(
-    directories: Union[Text, List[Text]]
-) -> Tuple[Set[Text], Set[Text]]:
+def _get_core_nlu_files(paths: Union[Text, List[Text]]) -> Tuple[Set[Text], Set[Text]]:
     story_files = set()
     nlu_data_files = set()
 
-    if isinstance(directories, str):
-        directories = [directories]
+    if isinstance(paths, str):
+        paths = [paths]
 
-    for directory in set(directories):
-        if not directory:
+    for path in set(paths):
+        if not path:
             continue
 
-        new_story_files, new_nlu_data_files = _find_core_nlu_files_in_directory(
-            directory
-        )
+        if os.path.isfile(path):
+            if _is_nlu_file(path):
```

Comment:
I guess, we should add the following check (as in `_find_core_nlu_files_in_directory`:
````
if not path.endswith(".json") and not path.endswith(".md"):
    continue
```
Maybe you can also extract the common part to a separate method.